src/App.js

```diff
@@ -1,16 +1,22 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
 import AddTodoForm from "./AddTodoForm";
 import TodoList from "./TodoList";
-
+const useSemiPersistentState =()=>{
+  // initialTodoList will get the stored list from localStorage if list dose not exist then it will be initialized with empty array 
+  const initialTodoList=JSON.parse(localStorage.getItem('savedTodoList'))==null?[]:JSON.parse(localStorage.getItem('savedTodoList'));
+  // State variable
+  const [todoList, setTodoList] = useState(initialTodoList)
+  // useEffect hook to store data in local storage 
+  useEffect(() => { 
+   localStorage.setItem('savedTodoList',JSON.stringify(todoList));
+  },[todoList]);
+  return [todoList,setTodoList];
+}
 function App() { 
-
-  const [todoList, setTodoList] = useState([]);
-
+  const[todoList,setTodoList]=useSemiPersistentState()
   let addTodo =(newTodo)=>{ 
-  //  console.log(newTodo);
     setTodoList([...todoList,newTodo])        
```

Comment:
So this is my fault -- After more research, I found the way we set the new array was wrong. React allows us to provide a function to the setter (setTodoList) and function param will get the most recent value of the state.

So it'd look something like this:

```suggestion
    setTodoList([...todoList,newTodo])        
    
     setTodoList( (previousTodoList) => [...previousTodoList, newTodo])
```
I used an arrow function there b/c it's easier to read. How does all this sound to you?

Reply:
Exercise 1.4 overview section actually explains my comment a bit better 😆
https://github.com/Code-the-Dream-School/react-cockatoo/wiki/Lesson-1.4
take a look

Reply:
thank you got it.